Automattic/newspack-plugin
# Fix/wizard admin menu priority 3

### All Submissions:






### Changes proposed in this Pull Request:

The [original wizards code](https://github.com/Automattic/newspack-plugin/blob/489ce0fb7390f9fabc5956844ac7b437c2c701ac/includes/wizards/class-wizard.php#L52) in `trunk` gave us the ability to adjust the priority of a wizard's `add_page` callback.  Somewhere in the timeline of the `epic/ia` development, this ability was hard-code then removed around the time we were fixing the parent menu custom ordering.  In the end, after [attempt 1](https://github.com/Automattic/newspack-plugin/pull/3534) and [attempt 2](https://github.com/Automattic/newspack-plugin/pull/3535), and lots of discussion, testing and a huddle with @miguelpeixe - we've settled on this PR in order to reinstate the removed code.

Changes:
- The parent menu order variable was renamed to be more descriptive of what it does: `$parent_menu_order`.
- The reinstated variable for add page priority was give a descriptive name: `$add_page_priority`.
- The old/hard-coded `newspack` parent menu was changed to `$this->parent_menu` so wizards can control which parent menu they are added to.
- Some minimal changes were made to the advertising wizards and the dashboard.

### How to test the changes in this Pull Request:

1. Pull this branch on top of `epic/ia`.
2. Run `npm run build` if you haven't already for `epic/ia`.
3. Verify the wp-admin menus are displaying as expected.

![parent-order](https://github.com/user-attachments/assets/1dfd8bb7-63d2-48a0-ac30-f4e2cf9bcfb4)

![advertising](https://github.com/user-attachments/assets/c6378caa-4879-4fd8-a40f-5db57c8bba65)

### Other information:
